dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/CoreSignInJourney.cs

```diff
@@ -1,47 +1,158 @@
-using Microsoft.AspNetCore.Mvc;
-using TeacherIdentity.AuthServer.Models;
-
 namespace TeacherIdentity.AuthServer.Journeys;
 
 public class CoreSignInJourney : SignInJourney
 {
-    public CoreSignInJourney(HttpContext httpContext, IdentityLinkGenerator linkGenerator)
-        : base(httpContext, linkGenerator)
+    public CoreSignInJourney(
+        HttpContext httpContext,
+        IdentityLinkGenerator linkGenerator,
+        CreateUserHelper createUserHelper)
+        : base(httpContext, linkGenerator, createUserHelper)
     {
     }
 
-    public override bool CanAccessStep(string step)
+    public override string GetNextStepUrl(string currentStep)
     {
-        throw new NotImplementedException();
-    }
+        if (IsFinished())
+        {
+            return currentStep switch
+            {
+                Steps.EmailConfirmation => GetStepUrl(Steps.EmailExists),
+                Steps.PhoneConfirmation => GetStepUrl(Steps.PhoneExists),
+                _ => AuthenticationState.PostSignInUrl
```

Comment:
We need to load the EmailExists and PhoneExists pages despite the Journey being completed